Helper/MerchantMetrics.php

```diff
@@ -0,0 +1,343 @@
+     *
+     *
+     * @throws
+     */
+    public function __construct(
+        Context $context,
+        ConfigHelper $configHelper,
+        DirectoryList $directoryList,
+        StoreManagerInterface $storeManager,
+        Bugsnag $bugsnag,
+        LogHelper $logHelper
+    ) {
+        parent::__construct($context);
+
+        $this->storeManager = $storeManager;
+        $this->bugsnag = $bugsnag;
+        $this->logHelper = $logHelper;
+        //////////////////////////////////////////
+        // Composerless installation.
+        // Make sure libraries are in place:
+        // lib/internal/Bolt/bugsnag
+        // lib/internal/Bolt/guzzle
+        //////////////////////////////////////////
+        if (!class_exists('\GuzzleHttp\Client')) {
+            // phpcs:ignore
+            require_once $directoryList->getPath('lib_internal') . '/Bolt/guzzle/autoloader.php';
+        }
+        $this->configHelper = $configHelper;
+        $this->metrics = array();
+        $this->metricsFile = null;
+        $this->guzzleClient = null;
+        $this->metricsFile = null;
+
+
+    }
+
+    /**
+     * Attempts to lock a file and returns a boolean based off of the result
+     *
+     * @param Stream        $workingFile    file that is attempting to be written to
+     *
+     * @return bool
+     */
+    protected function lockFile($workingFile){
+        return flock($workingFile, LOCK_EX | LOCK_NB);
+    }
+
+    /**
+     * Unlocks a file when finished
+     *
+     * @param Stream        $workingFile    file that is attempting to be written to
+     *
+     * @return void
+     */
+    protected function unlockFile($workingFile){
+        flock($workingFile, LOCK_UN);    // release the lock
+    }
+
+
+    /**
+     * Retrieves currrent time for when metrics are uploaded
+     *
+     * @return int
+     */
+    protected function getCurrentTime() {
+        return round(microtime(true) * 1000);
+    }
+
+    /**
+     * Based off the environment the project is ran in, determines the endpoint to add merchant metrics
+     *
+     * @return \GuzzleHttp\Client
+     */
+    protected function setClient() {
+        // determines if we are in the Sandbox env or not
+        $base_uri =  $this->configHelper->isSandboxModeSet() ? 'https://api-sandbox.bolt.com/' : 'https://api.bolt.com/' ;
+
+        // Creates a Guzzle Client and Headers needed for Metrics Requests
+        return new \GuzzleHttp\Client(['base_uri' => $base_uri]);
+    }
+
+    /**
+     * Retrieves Headers needed to communicate with Bolt
+     *
+     * @return array
+     */
+    protected function setHeaders() {
+        return [
+            'Content-Type' => 'application/json',
+            'x-api-key' =>  $this->configHelper->getApiKey()
+        ];
+    }
+
+    /**
+     * Fetches the current root directory for magento and creates a bolt metrics file in that location
+     *
+     * @return string
+     */
+    protected function setFile() {
```

Comment:
does this actually create file? it looks like it just returns the path of file; if so let's rename to getFilePath